### PR TITLE
OGIP19: Remove private dossiers' inactive status

### LINKED ISSUE #3570: Private Dossiers status `storniert` entfernen
Todos: 
 - Status inkl. Transitions entfernen aus 
 - Upgradestep für bestehende stornierte Inhalte
 - Upgradestep für Workflow Änderung

### PR DESCRIPTION
**Pull request by tarnap**
upgrade state vom inactive to active

Resolves #3570

### CODE REVIEW

**Comment by tarnap** [comment]
@phgross: I will squash the commits together when the PR is approved.

File: docs/HISTORY.txt [hunk 1]
```diff
@@ -5,6 +5,7 @@ Changelog
 2017.6.5 (unreleased)
 ---------------------
 
+- SPV: Remove deactivate action for private dossiers. [tarnap]
```
Comment (deiferni) [inline docs/HISTORY.txt]: this is not an SPV feature/issue 😉

Comment (tarnap) [reply docs/HISTORY.txt]: Indeed! I'll correct it! Thank you!

**Comment by tarnap** [comment]
@deiferni: Is it better this way? :)

File: opengever/core/upgrades/20171113145923_remove_deactivate_state_for_private_dossiers/upgrade.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+        deactivated_private_dossiers = self.catalog_unrestricted_search(query, full_objects=True)
+
+        state_mapping = {
+            ('opengever_private_dossier_workflow', 'opengever_private_dossier_workflow'): {
+                'dossier-state-inactive': 'dossier-state-active',
+            }
+        }
+
+        with WorkflowChainUpdater(deactivated_private_dossiers, state_mapping):
```
Comment (phgross) [inline opengever/core/upgrades/20171113145923_remove_deactivate_state_for_private_dossiers/upgrade.py]: The `WorkflowChainUpdater` does way to much in my opinion, a simple loop should be enough.
```
for obj in self.objects(query, 'Migrate deactivated private dossiers'):
    wf_tool.setStatusOf(wf_after, obj, {
                    'review_state': 'dossier-state-active',
                    'action': 'systemupdate',
                    'actor': 'system',
                    'comments': '',
                    'time': DateTime()})    
```

**Comment by tarnap** [comment]
@phgross: https://github.com/4teamwork/opengever.core/pull/3626/commits/3c37ce89252a3d0941d28479ce5e9c9314d403af << does not work yet:

### Before upgrade step
deactivated_before_upgrade_step

### After upgrade step
deactivated_after_upgrade_step

**Comment by tarnap** [comment]
@phgross: The upgrade step works now. This commit fixed it: https://github.com/4teamwork/opengever.core/pull/3626/commits/a36be8505115fdac9f36f96b690d3121f8338b3d

Let me know which commits you want me to squash together please :)

**Comment by phgross** [comment]
@tarnap please rebase and squash the seond and all following commits together ...

**Comment by tarnap** [comment]
@phgross: Rebased and commits merged.

**Pull request merged**